packages/manager/src/features/CloudPulse/Alerts/CreateAlert/CreateAlertDefinition.tsx

```diff
@@ -0,0 +1,245 @@
+  const { mutateAsync } = useCreateAlertDefinition();
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: engineOptions } = useDatabaseEnginesQuery(true);
+
+  const formik = useFormik({
+    initialValues,
+    onSubmit(
+      values: CreateAlertDefinitionPayload,
+      { resetForm, setErrors, setStatus, setSubmitting }
+    ) {
+      setStatus(undefined);
+      setErrors({});
+      const payload = { ...values };
+
+      mutateAsync(payload)
+        .then(() => {
+          setSubmitting(false);
+          enqueueSnackbar(`Alert created`, {
+            variant: 'success',
+          });
+          resetForm();
+          history.goBack();
+        })
+        .catch((err: APIError[]) => {
+          const mapErrorToStatus = () =>
+            setStatus({ generalError: getErrorMap([], err).none });
+          setSubmitting(false);
+          handleFieldErrors(setErrors, err);
+          handleGeneralErrors(mapErrorToStatus, err, 'Error creating an alert');
+        });
+    },
+    validationSchema: createAlertDefinitionSchema,
+  });
+
+  const {
+    errors,
+    handleBlur,
+    handleChange,
+    handleSubmit,
+    isSubmitting,
+    setFieldValue,
+    status,
+    values,
+  } = formik;
+
+  const generalError = status?.generalError;
+  const history = useHistory();
+  const onCancel = () => {
+    history.goBack();
+  };
+
+  const generateCrumbOverrides = (pathname: string) => {
+    const pathParts = pathname.split('/').filter(Boolean);
+    const lastTwoParts = pathParts.slice(-2);
+    const fullPaths: any = [];
+
+    pathParts.forEach((_, index) => {
+      fullPaths.push('/' + pathParts.slice(0, index + 1).join('/'));
+    });
+
+    const overrides = lastTwoParts.map((part, index) => ({
+      label: part,
+      linkTo: fullPaths[pathParts.length - 2 + index],
+      position: index + 1,
+    }));
+
+    return { newPathname: '/' + lastTwoParts.join('/'), overrides };
+  };
+
+  const { newPathname, overrides } = React.useMemo(
+    () => generateCrumbOverrides(location.pathname),
+    []
+  );
+
+  const CustomErrorMessage = (props: any) => (
+    <Box sx={(theme) => ({ color: theme.color.red })}>{props.children}</Box>
+  );
+
+  return (
+    <Paper>
+      <Breadcrumb
+        crumbOverrides={overrides}
+        pathname={newPathname}
+      ></Breadcrumb>
+      <FormikProvider value={formik}>
+        <form onSubmit={handleSubmit}>
+          {generalError && (
+            <Notice
+              data-qa-error
+              key={status}
+              text={status?.generalError ?? 'An unexpected error occurred'}
+              variant="error"
+            />
+          )}
+          <Typography variant="h2">1. General Information</Typography>
+          <TextField
+            label="Name"
+            name={'name'}
+            onBlur={handleBlur}
+            onChange={handleChange}
+            value={values.name ?? ''}
+          />
+          {formik.touched && formik.touched.name && errors.name ? (
```

Comment:
Lot of computations happening, inside return statement, better to move to a function